web/src/app/components/timeline/timeline.component.ts

```diff
@@ -0,0 +1,431 @@
+import {
+  AfterViewInit,
+  Component,
+  ElementRef,
+  Input,
+  ViewChild
+} from '@angular/core';
+import { formatDate } from '@angular/common';
+import {
+  HumanizeDurationLanguage,
+  HumanizeDuration,
+  HumanizeDurationOptions
+} from 'humanize-duration-ts';
+import * as d3 from 'd3';
+
+import { step189_2020 } from '../../../proto/step189_2020';
+
+/**
+ * Item holds all required data for one interval on the timeline.
+ */
+interface Item {
+  pushID: string;     // Push ID
+  state: number;      // Final state of push
+  startTime: number;  // Start of push, in milliseconds
+  endTime: number;    // End of push, in milliseconds
+  row: number;        // Row number corresponds to y-position on timeline
+}
+
+type d3SVG = d3.Selection<SVGSVGElement, Item[], null, undefined>;
+
+@Component({
+  selector: 'app-timeline',
+  templateUrl: './timeline.component.html',
+  styleUrls: ['./timeline.component.scss']
+})
+
+export class TimelineComponent implements AfterViewInit {
+  private static readonly LANG_SERVICE: HumanizeDurationLanguage
+    = new HumanizeDurationLanguage();
```

Comment:
The `=` should be on line 38 according to [1]. Some comment for the other similar places.

[1] https://google.github.io/styleguide/jsguide.html#formatting-where-to-break